direct/nn/openvino/openvino_model.py

```diff
@@ -11,37 +11,58 @@ def __init__(self, model):
+    def create_net(self, **kwargs):
         ie = IECore()
         ie.add_extension(get_extensions_path(), "CPU")
 
-        buf = io.BytesIO()
-        torch.onnx.export(
-            self.model,
-            (input_image, masked_kspace, sampling_mask, sensitivity_map),
-            buf,
-            opset_version=11,
-            operator_export_type=torch.onnx.OperatorExportTypes.ONNX_FALLTHROUGH,
-            input_names=["input_image", "masked_kspace", "sampling_mask", "sensitivity_map"],
-            output_names=["cell_outputs", "previous_state"],
-        )
-
-        net = ie.read_network(buf.getvalue(), b"", init_from_buffer=True)
-        self.exec_net = ie.load_network(net, "CPU")
-
-    def forward(self, input_image, masked_kspace, sampling_mask, sensitivity_map):
-        input_map = {
-            "input_image": input_image,
-            "masked_kspace": masked_kspace,
-            "sampling_mask": sampling_mask,
-            "sensitivity_map": sensitivity_map,
-        }
+        if self.model_name == "RIM":
+            args = ["input_image", "masked_kspace", "sampling_mask", "sensitivity_map"]
+            output_names = ["cell_outputs", "previous_state"]
+        elif self.model_name == "Unet2d":
+            output_names = ["output"]
+            if self.model.image_initialization == "sense":
+                args = ["masked_kspace", "sensitivity_map"]
+            elif self.model.image_initialization == "zero_filled":
+                args = ["masked_kspace"]
+            else:
+                raise ValueError(
+                    f"Unknown image_initialization. Expected `sense` or `zero_filled`. "
+                    f"Got {self.model.image_initialization}."
+                )
 
-        if self.exec_net is None:
-            self.create_net(input_image, masked_kspace, sampling_mask, sensitivity_map)
+        if args:
+            self.input = [kwargs[k] for k in args]
+        else:
+            raise ValueError(f"The model is not supported by OpenVINO: {self.model.__class__}")
+
+        with torch.no_grad():
+            buf = io.BytesIO()
+            torch.onnx.export(
+                self.model,
+                tuple(self.input),
+                buf,
+                opset_version=12,
+                operator_export_type=torch.onnx.OperatorExportTypes.ONNX_FALLTHROUGH,
+                input_names=args,
+                output_names=output_names,
+            )
 
-        res = self.exec_net.infer(input_map)
-        out = ([torch.Tensor(res["cell_outputs"])], torch.Tensor(res["previous_state"]))
+            net = ie.read_network(buf.getvalue(), b"", init_from_buffer=True)
+            self.exec_net = ie.load_network(net, "CPU")
 
+    def postprocess(self, res):
+        if self.model_name == "RIM":
+            out = ([torch.Tensor(res["cell_outputs"])], torch.Tensor(res["previous_state"]))
+        elif self.model_name == "Unet2d":
+            out = torch.Tensor(res["output"])
```

Comment:
Save output names into field and just iterate over them

Reply:
Can't use this approach because outputs has different types. RIM returns `(list of Tensors, Tensor)`, Unet - just `Tensor`